crates/sui-types/src/message_envelope.rs

```diff
@@ -291,7 +294,8 @@ where
     T: Message + UnauthenticatedMessage + Serialize,
 {
     pub fn verify_authority_signatures(&self, committee: &Committee) -> SuiResult {
-        self.data.verify_epoch(self.auth_sig().epoch)?;
+        // is this correct?
+        self.data.verify_epoch(self.auth_sig().epoch, true)?;
         self.auth_signature
```

Comment:
@mystenmark checking, iiuc i would not have access to the protocol config here right? is this ok just to pass in true here?

Reply:
if its not changing the behavior of existing code, then it should be fine. But this code is getting very confusing - instead of having a bool param, can we have separate `verify_epoch` and `verify_epoch_and_max_epoch` methods?